### PR TITLE
Fix deadlinks in docs

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Ran [deadlink](https://github.com/nschloe/deadlink) on Lightning docs

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: docs/source/advanced/advanced_gpu.rst [hunk 1]
```diff
@@ -117,7 +117,7 @@ To activate parameter sharding, you must wrap your model using provided ``wrap``
 When not using Fully Sharded these wrap functions are a no-op. This means once the changes have been made, there is no need to remove the changes for other plugins.
 
 ``auto_wrap`` will recursively wrap `torch.nn.Modules` within the ``LightningModule`` with nested Fully Sharded Wrappers,
-signalling that we'd like to partition these modules across data parallel devices, discarding the full weights when not required (information `here <https://fairscale.readthedocs.io/en/latest/api/nn/fsdp_tips.html>`__).
+signalling that we'd like to partition these modules across data parallel devices, discarding the full weights when not required (information `here <https://fairscale.readthedocs.io/en/latest/api/nn/fsdp.html>`__).
```
Comment (ethanwharris) [inline docs/source/advanced/advanced_gpu.rst:120]: We could add fairscale to the intersphinx configuration instead to resolve these automatically

**Review by awaelchli — APPROVED** [review]

File: docs/source/advanced/advanced_gpu.rst (hunk 1 above)
Comment (kaushikb11) [reply docs/source/advanced/advanced_gpu.rst:120]: @ethanwharris I have added `fairscale` and `graphcore` to `intersphinx_mapping`. Not sure how to add reference certain sections of their docs.

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by Borda — APPROVED** [review]
thinking can we have a job that would test all ink to be active?

**Review by ethanwharris — COMMENTED** [review]
@kaushikb11 Added some suggestions for fairscale that _should_ work. Don't think it can be done for graphcore as those links aren't listed in the intersphinx inventory on their docs :/
